### PR TITLE
revert PFM reversions. Disable PFM for ibcswap 

### PR DESCRIPTION
**Pull request by jcompagni10**
This undoes the reversions to PFM. 
It also disables the use of PFM in conjunction with ibcswap middleware. 
It also includes a small fix to ensure that unused tokens from swap are returned to the creator

### CODE REVIEW

File: tests/ibc/gmp_swap_forward_test.go [hunk 1]
```diff
@@ -88,31 +64,130 @@ func (s *IBCTestSuite) TestGMPSwapAndForward_Success() {
...
+	overrideAddr := s.ReceiverOverrideAddr(s.neutronTransferPath.EndpointA.ChannelID, s.providerAddr.String())
+	s.assertNeutronBalance(overrideAddr, s.providerToNeutronDenom, math.ZeroInt())
+	s.assertNeutronBalance(overrideAddr, s.providerToNeutronDenom, math.ZeroInt())
+
+	// Check that the unused balance is credited to the original creator
+	s.assertNeutronBalance(s.neutronAddr, s.providerToNeutronDenom, math.OneInt())
 }
+
+// TODO: fully validate PFM post ibcswap before re-enabling this
+// // TestSwapAndForward_Success asserts that the swap and forward middleware stack works as intended with Neutron running as a
```
Comment (mmikev) [inline tests/ibc/gmp_swap_forward_test.go]: Let's remove commented code

File: tests/ibc/swap_forward_test.go [hunk 2]
```diff
@@ -98,392 +87,604 @@ func (s *IBCTestSuite) TestSwapAndForward_Success() {
...
-	transferDenomPath := transfertypes.GetPrefixedDenom(
-		transfertypes.PortID,
-		s.neutronChainBPath.EndpointA.ChannelID,
-		nativeDenom,
-	)
-	transferDenomNeutronChainB := transfertypes.ParseDenomTrace(transferDenomPath).IBCDenom()
-
-	s.assertChainBBalance(chainBAddr, transferDenomNeutronChainB, math.ZeroInt())
-}
+// TODO: fully validate PFM post ibcswap before re-enabling this
```
Comment (mmikev) [inline tests/ibc/swap_forward_test.go]: Let's remove commented code

**Comment by jcompagni10** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/6933691110

**Comment by pr0n00gler** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/6990372127

**Pull request merged**